### PR TITLE
Merge keyframes at the same offset

### PR DESCRIPTION
**Pull request by hiikezoe**
This is a PR for https://bugzilla.mozilla.org/show_bug.cgi?id=1354947 .
---
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

- [X] These changes do not require tests because it's for stylo

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16850)

### CODE REVIEW

**Comment by hiikezoe** [comment]
Patches have been reviewed by @birtles and @SimonSapin .
@bors-servo r=birtles, SimonSapin

**Comment by hiikezoe** [comment]
@bors-servo r-
I can't make this PR merged today.

File: ports/geckolib/glue.rs [hunk 1]
```diff
@@ -2186,73 +2188,154 @@ pub extern "C" fn Servo_AssertTreeIsClean(root: RawGeckoElementBorrowed) {
...
+        (*keyframe).mPropertyValues.set_len((index + 1) as u32);
+        (*keyframe).mPropertyValues[index].mProperty = property.into();
+        // FIXME. Bug 1360398: Do not set computed values once we handles
+        // missing keyframes with additive composition.
+        (*keyframe).mPropertyValues[index].mServoDeclarationBlock.set_arc_leaky(
+            Arc::new(shared_lock.wrap(block)));
+    }
+}
+
+fn fill_in_missing_keyframe_values(all_properties:  &Vec<TransitionProperty>,
```
Comment (emilio) [inline ports/geckolib/glue.rs]: nit: This can just be `&[TransitionProperty]`.

Comment (hiikezoe) [inline ports/geckolib/glue.rs]: Neat! Thank you as always!

**Comment by hiikezoe** [comment]
Now there is no queue. It's time to merge!
@bors-servo r=birtles,SimonSapin

**Pull request merged**